src/Umbraco.Web.BackOffice/Middleware/BasicAuthAuthenticationMiddleware.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.Services;
+using Umbraco.Cms.Web.BackOffice.Security;
+using Umbraco.Extensions;
+
+namespace Umbraco.Cms.Web.Common.Middleware
+{
+    /// <summary>
+    ///     Ensures that preview pages (front-end routed) are authenticated with the back office identity appended to the
+    ///     principal alongside any default authentication that takes place
+    /// </summary>
+    public class BasicAuthAuthenticationMiddleware : IMiddleware
+    {
+        private readonly IRuntimeState _runtimeState;
+        private readonly IBasicAuthService _basicAuthService;
+
+        public BasicAuthAuthenticationMiddleware(
+            IRuntimeState runtimeState,
+            IBasicAuthService basicAuthService)
+        {
+            _runtimeState = runtimeState;
+            _basicAuthService = basicAuthService;
+        }
+
+        /// <inheritdoc />
+        public async Task InvokeAsync(HttpContext context, RequestDelegate next)
+        {
+            if (_runtimeState.Level < RuntimeLevel.Run || !_basicAuthService.IsBasicAuthEnabled())
+            {
+                await next(context);
+                return;
+            }
+
+            var clientIPAddress = context.Connection.RemoteIpAddress;
```

Comment:
Wondering if you need to jump through more hoops here to ensure you get the client's IP when accessing through a CDN?  E.g. something like [this](https://stackoverflow.com/a/66936819/489433), considering anything else that might be custom for common CDN providers.

Reply:
I don't think this is necessary. if the setup is correct.. Back in the days you added another middleware that handled x-forwarded-for etc.

Now it looks even more simple: https://devblogs.microsoft.com/aspnet/forwarded-headers-middleware-updates-in-net-core-3-0-preview-6/#configuration-only-wire-up-in-preview-6